sos/collector/clusters/saltstack.py

```diff
@@ -0,0 +1,88 @@
+# Copyright Red Hat 2022, Trevor Benson <trevor.benson@gmail.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import json
+from shlex import quote
+from sos.collector.clusters import Cluster
+
+
+class saltstack(Cluster):
+    """
+    The saltstack cluster profile is intended to be used on saltstack
+    clusters (Salt Project).
+    """
+
+    cluster_name = "Saltstack"
+    packages = ("salt-master",)
+    sos_plugins = ["saltmaster"]
+    cmd = "salt-run --out=pprint manage.status"
+    strict_node_list = True
+    option_list = [
+        ("compound", "", "Filter node list to those matching compound"),
+        ("glob", "", "Filter node list to those matching glob pattern"),
+        ("grain", "", "Filter node list to those with matching grain"),
+        ("list", "", "Filter node list to those matching list"),
+        ("minion_id_resolvable", False, "Returns minion IDs in the node list"
+         " for transports other than saltstck. Otherwise only nodes which are"
+         " up will have the fqdn grain returned."),
+        ("nodegroup", "", "Filter node list to those matching nodegroup"),
+        ("pillar", "", "Filter node list to those with matching pillar"),
+        ("regex", "", "Filter node list to those matching regex"),
+        ("subnet", "", "Filter node list to those in subnet"),
+    ]
+    targeted = False
+
+    def _parse_manage_status(self, output: str) -> list:
+        nodes = []
+        salt_json_output = json.loads(output.replace("'", '"'))
+        for _, value in salt_json_output.items():
+
```

Comment:
Why this empty line?

Also, why simple

    nodes = list(salt_json_output.values())

can't work - is there some gotcha I dont see?

Reply:
I didn't test list of lists for validity. 

### Nodes based on the current code
```
['store-2', 'store-3', 'store-4', 'store-5', 'store-0', 'store-1', 'supervisor']
```

### Nodes based on the suggested change 

#### When two keys exist, one for up and one for down nodes:
```
[['store-2', 'store-3', 'store-4', 'store-5'], ['store-0', 'store-1', 'supervisor']]
```

#### When only one key exists as all nodes are up (or down):
```
[['store-0', 'store-1', 'supervisor', 'store-2', 'store-3', 'store-4', 'store-5']]
```


If a list of lists is valid then I'll simplify and remove the for loop.

Reply:
Ah I see, thanks for explanation.